src/Package.php

```diff
@@ -129,17 +146,24 @@ private function deduceTrackingCode(): Package
         $tracking_code = $this->tracking_code;
         $carrier_code  = null;
         $provider_code = null;
+        $web_link = null;
 
         if (preg_match('/^[0-9]{2}[0-9]{4}[0-9]{4}$/', $tracking_code, $matches)) {
             $carrier_code = Carrier::CODE_DHL;
-        } elseif (preg_match('/^[1-9]{4}[0-9]{4}[0-9]{4}$/', $tracking_code, $matches)) {
+            $web_link = "https://www.dhl.com/en/express/tracking.html?AWB=".$matches[0];
```

Comment:
Hi @bell2041, thanks for the updates. My last piece of major feedback is that I think the Carrier class would be a better place for this tracking url logic to live.

For example, we could have a $tracking_urls mapping array property in the Carrier class similar to the carrier names: https://github.com/dsposito/argo/blob/master/src/Carrier.php#L54

Then we could set the value of a new Carrier $tracking_url property within the construct using the mapping array (and might need something like a buildTrackingUrl() helper method to insert the tracking code into the right place within the url). We could call that method within the construct(): https://github.com/dsposito/argo/blob/master/src/Carrier.php#L76

Let me know if that makes sense!

Reply:
Can you also add tests for the tracking urls once completed?